src/test/java/com/algorand/algosdk/integration/Stepdefs.java

```diff
@@ -526,21 +594,48 @@ public void msigNotInWallet()throws com.algorand.algosdk.kmd.client.ApiException
     }
 
     @When("I generate a key using kmd")
-    public void genKeyKmd() throws com.algorand.algosdk.kmd.client.ApiException, NoSuchAlgorithmException{
+    public void genKeyKmd() throws com.algorand.algosdk.kmd.client.ApiException, NoSuchAlgorithmException {
         GenerateKeyRequest req = new GenerateKeyRequest();
         req.setDisplayMnemonic(false);
         req.setWalletHandleToken(handle);
         pk = new Address(kcl.generateKey(req).getAddress());
     }
 
+    @When("I generate a key using kmd for rekeying")
+    public void genKeyKmdRekey() throws com.algorand.algosdk.kmd.client.ApiException, NoSuchAlgorithmException {
+        GenerateKeyRequest req = new GenerateKeyRequest();
+        req.setDisplayMnemonic(false);
+        req.setWalletHandleToken(handle);
+        rekey = new Address(kcl.generateKey(req).getAddress());
+
+        // Fund rekey address
```

Comment:
@winder Leaving context below and resolving (since PR is approved).  Happy to discuss further.

You're correct that in isolation the address need _not_ be funded.  However, one of the existing tests issues a transaction with the rekeyed address (attached logs - [output.txt](https://github.com/algorand/java-algorand-sdk/files/9212069/output.txt)).

An alternative is to design a step implementations specific to rekeying + transaction that avoid generally funding the rekey account.  

My preference is to leave as is for the PR because I'm attempting to minimize change set size.

Reply:
What do you think about naming the new step "I generate a key using kmd for rekeying and fund it"?

Reply:
@winder Sounds reasonable - I brought in the feedback via https://github.com/algorand/algorand-sdk-testing/pull/206/commits/7992e0ac032293a0e4fdd5c48b23f2a0449f204e and https://github.com/algorand/java-algorand-sdk/pull/350/commits/c0cd66fb32e7c3ab660467f900a7bc1ba3156f8e.

If it looks good, can you resolve the thread?

Reply:
ty, resolving.